src/structures/member.ts

```diff
@@ -60,39 +71,32 @@ const baseMember: Partial<Member> = {
 };
 
 export async function createMemberStruct(
-  data: MemberCreatePayload,
+  data: Omit<DiscordGuildMember, "user"> & { user: DiscordUser },
```

Comment:
Add a comment explaining why this is done for future reference

Reply:
Is this good?
`DiscordUser contains important informations like: id, username and discriminator which are required to build this structure`

Reply:
Not why it is required, but why it is non-optional in this case and optional in the interface.

Reply:
The `user` param in `DiscordGuildMember` is optional since discord does not send it in `MESSAGE_CREATE` and `MESSAGE_UPDATE` events. But this data in there is required to build this structure so it is required in this case?